tree_queries/compiler.py

```diff
@@ -152,39 +163,42 @@ def get_sibling_order_params(self):
             raise ValueError(
                 "Sibling order must be a string or a list or tuple of strings."
             )
-
-        # Use Django to make a SQL query whose parts can be repurposed for __rank_table
-        base_query = (
-            _find_tree_model(self.query.model)
-            .objects.only("pk", "parent")
-            .order_by(*order_fields)
-            .query
+        
+        # Convert strings to expressions. This is to maintain backwards compatibility
+        # with Django versions < 4.1
```

Comment:
Would it be possible to remove much code when we remove support for Django before 4.2? I'm an unfortunate user of Django 4.0 still, but 4.2 is the oldest version of Django which is still maintained and I think we could move ahead if it simplifies the code (see https://www.djangoproject.com/download/)

Reply:
We could only remove the block marked out by comments which converts order strings to `F` expressions (~12 lines). In 4.1 Django changed to order_by field in window expressions so that it would accept strings like the queryset order_by function, prior to that all ordering fields needed to be in the form of an expression object.

Reply:
Ah OK. I'm asking for a small change then: I generally try to avoid `VERSION` comparisons, but in this case it would be helpful to use something like `if django.VERSION < (4, 2)` since [django-upgrade is able to automatically remove such blocks from the code](https://github.com/adamchainz/django-upgrade?tab=readme-ov-file#versioned-blocks)

Reply:
I did not know that about django-upgrade. A version check has been added!